samples: bluetooth: Fix periodic_sync_rsp sample

olivier-le-sage opened the pull request:
Was broken after BT_LE_ADV_OPT_USE_NAME and
BT_LE_ADV_OPT_FORCE_NAME_IN_AD were deprecated.

The periodic_adv_rsp sample
looks for the string "PAwR sync sample" and connects based on that.

So, the periodic_sync_rsp sample needs to advertise this string.

olivier-le-sage committed:
samples: bluetooth: fix periodic_sync_rsp sample connection

Was broken after BT_LE_ADV_OPT_USE_NAME and
BT_LE_ADV_OPT_FORCE_NAME_IN_AD were deprecated.

The periodic_adv_rsp sample
looks for the string "PAwR sync sample" and connects based on that.

So, the periodic_sync_rsp sample needs to advertise this string.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
samples: bluetooth: minor logging changes in PAwR samples

Some minor logging improvements based on my user experience

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

Thalley commented on samples/bluetooth/periodic_sync_rsp/src/main.c:
```diff
@@ -229,7 +229,7 @@ int main(void)
 			BT_LE_ADV_PARAM(BT_LE_ADV_OPT_ONE_TIME | BT_LE_ADV_OPT_CONNECTABLE,
 					BT_GAP_ADV_FAST_INT_MIN_2, BT_GAP_ADV_FAST_INT_MAX_2, NULL),
-			NULL, 0, sd, ARRAY_SIZE(sd));
+			ad, ARRAY_SIZE(ad), NULL, 0);
```
So what you are doing here is using advertising data instead of scan response data for the name. 

That's fine, but either should work, assuming that the remote device are using active scan. 

I don't think it's right to call it broken, as this should still work, so I would suggest to reword the commit message.

Thalley replied:
The alternative is to modify `periodic_adv_rsp` to use active scan instead of `BT_LE_SCAN_PASSIVE_CONTINUOUS`

Thalley reviewed (dismissed):
Changes look good, but commit message should be updated. It's not broken, just incompatible with another sample. 

@theob-pro  I do not agree that this is a bug (the print statements fixes are though)

Thalley commented:
> > @theob-pro I do not agree that this is a bug (the print statements fixes are though)
> 
> IMO this is a bug. After I deprecated the auto name in ad, I updated the samples that were relying on this feature. I made a mistake in updating that sample that caused a change of behavior.

I see, yeah previously I guess the sample earlier had the name in the AD rather than SD, so it indeed a change in behavior, but I still wouldn't call it broken as any observer that is doing active scan would still get it correctly.

theob-pro commented:
> > > @theob-pro I do not agree that this is a bug (the print statements fixes are though)
> > 
> > 
> > IMO this is a bug. After I deprecated the auto name in ad, I updated the samples that were relying on this feature. I made a mistake in updating that sample that caused a change of behavior.
> 
> I see, yeah previously I guess the sample earlier had the name in the AD rather than SD, so it indeed a change in behavior, but I still wouldn't call it broken as any observer that is doing active scan would still get it correctly.

I agree but at the same time I guess people use the corresponding sample together. Anyway, that's not a big deal, if you want to remove the bug label feel free to do it :slightly_smiling_face:

olivier-le-sage commented:
> > > @theob-pro I do not agree that this is a bug (the print statements fixes are though)
> > 
> > 
> > IMO this is a bug. After I deprecated the auto name in ad, I updated the samples that were relying on this feature. I made a mistake in updating that sample that caused a change of behavior.
> 
> I see, yeah previously I guess the sample earlier had the name in the AD rather than SD, so it indeed a change in behavior, but I still wouldn't call it broken as any observer that is doing active scan would still get it correctly.

This isn't a scan/adv sample. It's meant to demonstrate periodic sync with responses. For this, a minimum of two devices are needed, and to make this simple, we split this into two main.c files with instructions in the readme saying the two applications are to be used together.

Without this change, it doesn't synchronize. It's not usable for its intended purpose; it is broken.

Thalley commented:
> Without this change, it doesn't synchronize. It's not usable for its intended purpose; it is broken.

I guess this is up to semantics. You are assuming that the purpose of this sample is to be run with the related sample. I do not share this assumption, as the purpose of a sample is to demonstrate some behavior, which this sample does correctly regardless. 

I'll dismiss my request for change because the README contains the following statement:

> Use the sample found under :zephyr_file:`samples/bluetooth/periodic_sync_rsp` in the Zephyr tree that will synchronize and respond to this sample.

which is rather unique for this set of samples, but I'll refrain from approving it until the commit message is updated (others can approve though). IMO the commit message should something along the lines of

> Moves the device name from scan response to advertising data so that devices using passive scan can receive the device name as well.

olivier-le-sage commented:
> You are assuming that the purpose of this sample is to be run with the related sample. 

I know the purpose of this sample because I developed the feature it demonstrates, and worked on the team that put this sample there to begin with. As you noticed, we documented our intentions in the README.rst files.

> The alternative is to modify periodic_adv_rsp to use active scan instead of BT_LE_SCAN_PASSIVE_CONTINUOUS

For this to work the periodic_adv_rsp sample needs to be modified to use BT_LE_SCAN_ACTIVE or similar, and to not ignore BT_GAP_ADV_TYPE_SCAN_RSP. The point of this initial advertisement is to create a connection, then use sync transfer. I chose not to go for this option to use scan requests / responses because it's not useful, and this is a bit simpler.
